### PR TITLE
Refine test cases to match query with filter

### PR DESCRIPTION
**Pull request by iAKM**

### CODE REVIEW

File: GoInfoGame/GoInfoGameTests/quests/TestQuestUtils.swift [hunk 1]
```diff
@@ -13,6 +13,14 @@ class TestQuestUtils {
         return Int64(Date().timeIntervalSince1970)
     }
     
+    static func olderThan(years: Int) -> Int64 {
```
Comment (susrisha) [inline GoInfoGame/GoInfoGameTests/quests/TestQuestUtils.swift]: rename to `addTime` and enhance the function to take months, days and years as input. Just years will not work out

File: GoInfoGame/GoInfoGameTests/quests/BusStopLitTests.swift [hunk 2]
```diff
@@ -41,13 +41,18 @@ final class BusStopLitTests: XCTestCase {
      */
     
     func testBusStopLitQuery() throws {
+        let eightYearsAgo = TestQuestUtils.olderThan(years: -8)
+        let sixteenYearsAgo = TestQuestUtils.olderThan(years: 8)
```
Comment (susrisha) [inline GoInfoGame/GoInfoGameTests/quests/BusStopLitTests.swift]: why is this advancing another 8 years? where is this used?

File: GoInfoGame/GoInfoGameTests/quests/StepsRampTests.swift [hunk 3]
```diff
@@ -40,15 +40,20 @@ final class StepsRampTests: XCTestCase {
      */
     
     func testStepsRampQuery() throws {
+        let fourYearsAgo = TestQuestUtils.olderThan(years: -4)
```
Comment (susrisha) [inline GoInfoGame/GoInfoGameTests/quests/StepsRampTests.swift]: both are same.. `fourYearsAgo` and `eightYearsAgo`. Why are two variables needed?

File: GoInfoGame/GoInfoGameTests/quests/TactilePavingStepsTests.swift [hunk 4]
```diff
@@ -44,13 +44,21 @@ final class TactilePavingStepsTests: XCTestCase {
...
         assertIsNotApplicable(element: TestQuestUtils.node(tags: ["highway": "residential"]))
-        assertIsApplicable(element: TestQuestUtils.way(tags: ["highway": "steps", "surface": "paved"]))
-        assertIsNotApplicable(element: TestQuestUtils.way(tags: ["highway": "steps", "tactile_paving": "yes", "tactile_paving:date": "2023-01-31"])) // tactile_paving older than 8 years
-        assertIsNotApplicable(element: TestQuestUtils.way(tags: ["highway": "steps", "tactile_paving": "partial", "tactile_paving:date": "2023-01-31"])) // tactile_paving older than 4 years
+        assertIsApplicable(element: TestQuestUtils.way(tags: ["highway" : "steps", "surface": "paved"]))
+        assertIsApplicable(element: TestQuestUtils.way(tags: ["highway" : "steps", "surface": "paved", "tactile_paving": "unknown"]))
+        assertIsApplicable(element: TestQuestUtils.way(tags: ["highway" : "steps", "surface": "paved", "tactile_paving": "incorrect"], timestamp: fourYearsAgo))
+        assertIsApplicable(element: TestQuestUtils.way(tags: ["highway" : "steps", "surface": "paved", "tactile_paving": "yes"], timestamp: eightYearsAgo))
```
Comment (susrisha) [inline GoInfoGame/GoInfoGameTests/quests/TactilePavingStepsTests.swift:60]: where are the negative cases for timestamps??

**Review by susrisha — CHANGES_REQUESTED** [review]

File: GoInfoGame/GoInfoGameTests/quests/TestQuestUtils.swift (hunk 1 above)
Comment (iAKM) [reply GoInfoGame/GoInfoGameTests/quests/TestQuestUtils.swift]: resolved

File: GoInfoGame/GoInfoGameTests/quests/BusStopLitTests.swift (hunk 2 above)
Comment (iAKM) [reply GoInfoGame/GoInfoGameTests/quests/BusStopLitTests.swift]: removed the unused variable

File: GoInfoGame/GoInfoGameTests/quests/StepsRampTests.swift (hunk 3 above)
Comment (iAKM) [reply GoInfoGame/GoInfoGameTests/quests/StepsRampTests.swift]: removed redundant variable

File: GoInfoGame/GoInfoGameTests/quests/TactilePavingStepsTests.swift (hunk 4 above)
Comment (iAKM) [reply GoInfoGame/GoInfoGameTests/quests/TactilePavingStepsTests.swift:60]: added negative cases for timestamps

**Review by susrisha — APPROVED** [review]

**Pull request merged**